### PR TITLE
Quick ore pick up + quick sandbag deployment

### PR DESCRIPTION
**Pull request by Super3222**
### Intent of your Pull Request

- You can now deploy sandbag barricades by clicking on floors instead of having to open the construction menu
- Walking over ore with a mining satchel will automatically pick it up. No, I did not port this from /tg/. Though I did hear about the system, so I thought I was already making a PR to clear up tedious shit miners have to go through so why not code it from scratch.

#### Changelog

:cl:
rscadd: You can now deploy sandbags easier! Try clicking on a floortile with a bag in your hand.
rscadd: You can now walk over ore with a mining satchel in hand to pick it up.
/:cl:

### CODE REVIEW

**Comment by LarissaMayrink** [comment]
Not that this is bad, but ripleys will become even slower than double hka at mining.

**Comment by Super3222** [comment]
> Not that this is bad, but ripleys will become even slower than double hka at mining.

Pretty good point, Larissa. That got me thinking... how about letting ripleys automatically pick up ores that are 1 tile away from them? As long as they have their clamp selected & their orebox already lifted into their mech.

**Comment by LarissaMayrink** [comment]
@Super3222 they already autopick whatever they mine with the drill, so, it won't help at all.

File: code/modules/mining/ores_coins.dm [hunk 1]
```diff
@@ -17,6 +17,19 @@
...
+		var/mob/living/carbon/human/H = AM
+		var/obj/item/weapon/storage/bag/ore/orestorage
+		if(istype(H.r_hand, /obj/item/weapon/storage/bag/ore))
+			orestorage = H.r_hand
+		else if(istype(H.l_hand, /obj/item/weapon/storage/bag/ore))
+			orestorage = H.l_hand
+		var/turf/T = get_turf(src)
+		for(var/obj/item/weapon/ore/O in T)
+			if(orestorage.can_be_inserted(O))
+				orestorage.handle_item_insertion(O, 1, user = H)
```
Comment (Cruix) [inline code/modules/mining/ores_coins.dm]: Check if the ore's loc is still the turf so you aren't inserting the ore multiple times

File: code/game/turfs/turf.dm [hunk 2]
```diff
@@ -52,6 +52,16 @@
...
+	if(istype(C, /obj/item/stack/sheet/mineral/sandbags))
+		var/obj/item/stack/sheet/mineral/sandbags/S = C
+		user << "<span class='notice'>You start building a barricade...</span>"
+		if(do_after(user, 30, target = src))
+			new /obj/structure/barricade/sandbags(get_turf(src))
+			S.use(1)
+			user << "<span class='notice'>You construct a sand barricade.</span>"
+		else
+			user << "<span class='notice'>You have been interrupted!</span>"
+		return 1
```
Comment (Cruix) [inline code/game/turfs/turf.dm]: Put this in the afterattack of sandbags and make it behave the same as constructing it from the menu

Comment (Super3222) [reply code/game/turfs/turf.dm]: > make it behave the same as constructing it from the menu

I can make the creation times the same, but beyond that I don't know. I'll make a check to make sure a sandbags not already onthat turf

Comment (Super3222) [reply code/game/turfs/turf.dm]: I don't understand why i have to put it in afterattack either, but okay.

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 3]
```diff
@@ -75,6 +76,27 @@ var/global/list/datum/stack_recipe/sandbag_recipes = list ( \
 	pixel_y = rand(0,4)-4
 	..()
 
+/obj/item/stack/sheet/mineral/sandbags/afterattack(atom/target, mob/user)
+	if(istype(target, /turf/open/floor))
+		var/turf/open/floor/F = target
+		var/occupied
+		for(var/obj/structure/barricade/sandbags/S in F)
+			if(occupied)
```
Comment (Cruix) [inline code/game/objects/items/stacks/sheets/mineral.dm:84]: This will never be true

**Comment by Cruix** [comment]
You know, there is a device in the vendor specifically designed to pick up ore as you walk over it.

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 4]
```diff
@@ -75,6 +76,27 @@ var/global/list/datum/stack_recipe/sandbag_recipes = list ( \
...
+
+		user << "<span class='notice'>You start building a barricade...</span>"
+		if(do_after(user, 25, target = F))
+			new /obj/structure/barricade/sandbags(get_turf(F))
+			use(1)
+			user << "<span class='notice'>You construct a sand barricade.</span>"
+		else
+			user << "<span class='notice'>You have been interrupted!</span>"
+		return 1
+
```
Comment (Cruix) [inline code/game/objects/items/stacks/sheets/mineral.dm:99]: return ..()

**Pull request closed without merging**